lib/location/rest/rest_services_wifi.h

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/**
+ * @file rest_services_wifi.h
+ *
+ * @brief REST API for WiFi positioning.
+ */
+
+#ifndef REST_SERVICES_WIFI_H
+#define REST_SERVICES_WIFI_H
+
+#include <zephyr/types.h>
+#include <net/wifi.h>
+#include <modem/location.h>
+
+#define REST_WIFI_MIN_TIMEOUT_MS 1500
+#define WIFI_MAC_ADDR_STR_LEN 17
+
+/** @brief Item for passing a WiFi scanning result */
+struct wifi_scanning_result_info {
+	char mac_addr_str[WIFI_MAC_ADDR_STR_LEN + 1];
+	char ssid_str[WIFI_SSID_MAX_LEN + 1];
+	uint8_t channel;
+	int8_t rssi;
+};
+
+/** @brief Data used for WiFi positioning request */
+struct rest_wifi_pos_request {
+	/** Scanning results */
+	struct wifi_scanning_result_info
+		scanning_results[CONFIG_LOCATION_METHOD_WIFI_SCANNING_RESULTS_MAX_CNT];
+	uint8_t wifi_scanning_result_count; /* Count of wifi access points returned in scanning */
+	int32_t timeout_ms; /* REST request timeout (in mseconds) */
+};
+
+/** @brief WiFi positioning request result */
+struct rest_wifi_pos_result {
```

Comment:
I'm thinking if we should use loc_location instead of defining this again here.

Reply:
started using `location_data`